src/test/java/org/opensearch/security/auditlog/integration/TestAuditlogImpl.java

```diff
@@ -58,30 +58,66 @@ public static synchronized void clear() {
      * Perform an action and then wait until the expected number of messages have been found.
      */
     public static List<AuditMessage> doThenWaitForMessages(final Runnable action, final int expectedCount) {
-        final CountDownLatch latch = new CountDownLatch(expectedCount);
+        final List<AuditMessage> missedMessages = new ArrayList<>();
         final List<AuditMessage> messages = new ArrayList<>();
-        countDownRef.set(latch);
-        messagesRef.set(messages);
-
-        TestAuditlogImpl.sb = new StringBuffer();
-        TestAuditlogImpl.messages = messages; 
+        final CountDownLatch latch = resetAuditStorage(expectedCount, messages);
         
         try {
```

Comment:
Thoughts on surrounding this all with one try block?

Reply:
Seems like a good nit, if I make any more changes I'll include this in with it - sounds fair?